hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/table/upgrade/TestUpgradeDowngrade.java

```diff
@@ -254,6 +293,20 @@ private static Stream<Arguments> upgradeDowngradeVersionPairs() {
     );
   }
 
+  /**
+   * Version pairs that should be tested for metadata upgrade/downgrade failures.
+   * Excludes pairs that trigger rollback operations which delete metadata tables.
+   */
+  private static Stream<Arguments> metadataTableCorruptionTestVersionPairs() {
+    return Stream.of(
+        Arguments.of(HoodieTableVersion.FOUR, HoodieTableVersion.FIVE),   // V4 -> V5
+        Arguments.of(HoodieTableVersion.FIVE, HoodieTableVersion.FOUR)    // V5 -> V4
+    // Note: Other pairs like V5->V6, V6->V8 are excluded because they may trigger
+    // rollback operations in rollbackFailedWritesAndCompact() that disable metadata
```

Comment:
why would it trigger deletion of MDT? 
Our test set up should be such that. 

testing 4 to 5: 
   create data table and mdt in v4. corrupt mdt. and try upgrade. 

testing 6 to 8:
   create data table and mdt in v8. corrupt mdt. try upgrade. 

given this test structure, why would mdt be deleted.

Reply:
Synced with @nsivabalan today on this. Currently the code within the rollbackAndCompact util method explicitly disabled the mdt table, which would eventually lead to the mdt table being gone.

https://github.com/apache/hudi/blob/master/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/UpgradeDowngradeUtils.java#L205

For now we are aligned on restricting the test case be for handlers that do not call rollbackAndCompact in order to test this metadata failure case